### PR TITLE
Add Schema for ACL

### PR DESCRIPTION
**Pull request by xdwu**
Add Schema for ACL:
- ACL_POLICY_TABLE
- ACL_RULE_TABLE
- ACL_INTERFACE_TABLE

### CODE REVIEW

**Comment by xdwu** [comment]
Open for Discussion

File: doc/swss-schema.md [hunk 1]
```diff
@@ -280,7 +280,41 @@ and reflects the LAG ports into the redis under: `LAG_TABLE:<team0>:port`
...
+  ip_src 	= IPprefix/"any"  			; options of the source ip address (and mask)
+  ip_dst	= IPprefix/"any"			; options of the destination ip address (and mask)
+  l4_src_port	= port_num/[port_num_L-port_num_H]	; L4 source port or the range of ports
+  l4_dst_port	= port_num/[port_num_L-port_num_H]	; L4 destination port or the range of ports
+  seq		= 1*DIGIT				; sequence number of the rules assocaited with this ACL policy
+  port_num	= 1*5DIGIT				; a number between 0 and 65535, port_num_L < port_num_H
+  							; eq port X: 	X
+							; gt port X:	[X-65535]
+							; lt port X; 	[0-X]
+							; range X, Y	[X-Y]
```
Comment (lguohan) [inline doc/swss-schema.md]: convert all tabs to spaces

File: doc/swss-schema.md [hunk 2]
```diff
@@ -280,7 +280,41 @@ and reflects the LAG ports into the redis under: `LAG_TABLE:<team0>:port`
...
 
+---------------------------------------------
+###ACL_INTERFACE_TABLE
+; Define the interfaces (ports) to which the ACL Policy is applied 
+; Only One ACL Policy can be applied to an interface
+; One ACL can be applied to multiple interfaces
+; Status: Working in progress
+  key: ACL_INTERFACE_TABLE:policy_name:port_name	; ACL policy defined in "policy_name" is applied to this interface 
+  port_name	= 1*64VCHAR 				; name of this port
```
Comment (lguohan) [inline doc/swss-schema.md]: why not add port_name in the acl_policy_table?

File: doc/swss-schema.md [hunk 3]
```diff
@@ -280,7 +280,41 @@ and reflects the LAG ports into the redis under: `LAG_TABLE:<team0>:port`
...
+###ACL_RULE_TABLE
+; Define rules associated with a specific ACL Policy
+; Status: Working in progress
+  key: ACL_RULE_TABLE:policy_name:seq			; seq is the index of the rules when the packet is filtered by the ACL "policy_name"
+  action	= "permit"/"deny"			; action when the fields are matched
+  protocol	= "ip"/"icmp"/"tcp"/"udp"/"any"		; options of the protocol field
+  ip_src 	= IPprefix/"any"  			; options of the source ip address (and mask)
+  ip_dst	= IPprefix/"any"			; options of the destination ip address (and mask)
+  l4_src_port	= port_num/[port_num_L-port_num_H]	; L4 source port or the range of ports
+  l4_dst_port	= port_num/[port_num_L-port_num_H]	; L4 destination port or the range of ports
```
Comment (lguohan) [inline doc/swss-schema.md]: these are fields.

add ;fields, see above for example

File: doc/swss-schema.md (hunk 1 above)
Comment (lguohan) [inline doc/swss-schema.md]: these are values, add

; values

see above for examples

File: doc/swss-schema.md (hunk 2 above)
Comment (xdwu) [inline doc/swss-schema.md]: Sure.

File: doc/swss-schema.md [hunk 4]
```diff
@@ -280,7 +280,56 @@ and reflects the LAG ports into the redis under: `LAG_TABLE:<team0>:port`
     ttl_mode                = "uniform" / "pipe"
 
 ---------------------------------------------
-
+###ACL_POLICY_TABLE
+; Define ACL Policy Table
+; Status: Working in progress
+  key           = ACL_POLICY_TABLE:policy_name          ; policy_name must be unique
```
Comment (lguohan) [inline doc/swss-schema.md]: add line ";field  value", see above example

File: doc/swss-schema.md [hunk 5]
```diff
@@ -280,7 +280,56 @@ and reflects the LAG ports into the redis under: `LAG_TABLE:<team0>:port`
...
+                                                        ; ACL policy might not be applied to any interface yet.
+  port_name     = 1*64VCHAR                             ; name of the port, must be unique
+  max_ports     = 1*5DIGIT                              ; number of ports supported on the chip
+---------------------------------------------
+###ACL_RULE_TABLE
+; Define rules associated with a specific ACL Policy
+; Status: Working in progress
+  key: ACL_RULE_TABLE:policy_name:seq                   ; key of the rule entry in the table, seq is the order of the rules 
+                                                        ; when the packet is filtered by the ACL "policy_name". 
+                                                        ; A rule is always assocaited with a policy.
```
Comment (lguohan) [inline doc/swss-schema.md]: add line ";field value", see above example

**Comment by lguohan** [comment]
close this one, will open another pr later

**Pull request closed without merging**